From 31a174054c86fd51e81185d5b2bfcf2ab6bcf208 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 1 Feb 2021 12:44:15 +0000
Subject: [PATCH] gnu: Add gopkg.in/yaml.v3

* gnu/packages/golang.scm (go-gopkg-in-yml-v3): New variable
---
 gnu/packages/golang.scm | 42 +++++++++++++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index d34bb1acb7..56843820f0 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -23,6 +23,7 @@
 ;;; Copyright © 2020 Marius Bakke <marius@gnu.org>
 ;;; Copyright © 2020 raingloom <raingloom@riseup.net>
 ;;; Copyright © 2020 Martin Becze <mjbecze@riseup.net>
+;;; Copyright © 2021 Sharlatan Hellseher <sharlatanus@mgail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -2408,6 +2409,47 @@ which satisfies the cron expression.")
 values.")
     (license license:asl2.0)))
 
+(define-public go-gopkg-in-yaml-v3
+  (package
+    (name "go-gopkg-in-yaml-v3")
+    (version "3")
+    (source
+      (origin
+        (method git-fetch)
+        (uri (git-reference
+               (url "https://gopkg.in/yaml.v3.git")
+               (commit (string-append "v" version))))
+        (file-name (git-file-name name version))
+        (sha256
+         (base32 "06f4lnrp494wqaygv09dggr2dwf3z2bawqhnlnnwiamg5y787k4g"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "gopkg.in/yaml.v3"))
+    (native-inputs
+     `(("go-gopkg-in-check-v1" ,go-gopkg-in-check-v1)))
+    (home-page "https://gopkg.in/yaml.v2")
+    (synopsis "YAML reader and writer for the Go language")
+    (description
+     "This package provides a Go library for encode and decode YAML
+values.
+
+v3 specification difference:
+@itemize
+@item YAML 1.1 bools (yes/no, on/off) are supported as long as they are
+being decoded into a typed bool value.  Otherwise they behave as a string.
+Booleans in YAML 1.2 are true/false only.
+
+@item Octals encode and decode as 0777 per YAML 1.1, rather than 0o777 as
+specified in YAML 1.2, because most parsers still use the old format.  Octals in
+the 0o777 format are supported though, so new files work.
+
+@item Does not support base-60 floats.  These are gone from YAML 1.2, and were
+actually never supported by this package as it's clearly a poor choice.
+@end itemize\n")
+
+@end itemize\n")
+    (license license:asl2.0)))
+
 (define-public go-github-com-mattn-go-isatty
   (package
     (name "go-github-com-mattn-go-isatty")
-- 
2.30.0

